step-plans/step-plans-core/src/main/java/step/core/artefacts/reports/ReportNodeAccessorImpl.java

```diff
@@ -76,7 +76,18 @@ public Iterator<ReportNode> getChildren(ObjectId parentID) {
 	public Iterator<ReportNode> getChildren(ObjectId parentID, int skip, int limit) {   
     	return collectionDriver.find(Filters.equals("parentID", parentID), new SearchOrder("executionTime", 1), skip, limit, 0).iterator();
     }
-    
+
+	@Override
+	public Iterator<ReportNode> getChildrenByParentSource(ObjectId parentID, ParentSource parentSource) {

```

Comment:
We should check the performance impact of this. Do we need new indexes?

Reply:
I will create a followup ticket for this as dicussed